cli/src/pcluster/templates/cw_dashboard_builder.py

```diff
@@ -219,6 +228,131 @@ def _add_conditional_storage_widgets(
+            custom_metric_filter(
+                error_information="Inability to launch due to limits",
+                metric_name="Volume Limit",
+                filter_pattern="VolumeLimitExceeded",
+            ),
+            custom_metric_filter(
+                error_information="Insufficient Capacity ",
+                metric_name="Node Capacity Insufficient",
+                filter_pattern="InsufficientInstanceCapacity",
+            ),
+        ]
+
+        custom_script_errors = [
+            custom_metric_filter(
+                error_information="Failure in retrieving custom actions",
+                metric_name="Cannot retrieve custom script",
+                filter_pattern="error occurred 403 when calling the HeadObject operation Forbidden",
+            ),
+            custom_metric_filter(
+                error_information="Error in Running Custom Actions",
+                metric_name="Error With Custom Script",
+                filter_pattern="Bootstrap failed with error Failed to run bootstrap recipes",
+            ),
+            custom_metric_filter(
+                error_information="Script Timeout",
+                metric_name="Infinite Custom Script",
+                filter_pattern="systemd udevd worker timeout kill it",
+            ),
+        ]
+
+        compute_node_termination_errors = [
+            custom_metric_filter(
+                error_information="EC2 termination of compute nodes out of PCluster control",
+                metric_name="Terminated EC2 compute node before job submission ",
+                filter_pattern="WARNING Node state check no corresponding instance in EC2 for node",
+            ),
+        ]
+
+        job_issues_errors = [
+            custom_metric_filter(
+                error_information="Instances failing EC2 health check",
+                metric_name="EC2 Health Check",
+                filter_pattern="WARNING Node state check no corresponding instance in EC2 for node",
+            ),
+            custom_metric_filter(
+                error_information="Instances in EC2 Maintenance Events",
+                metric_name="EC2 Maintenance Events",
+                filter_pattern="Setting nodes failing health check type ec2_health_check to DRAIN",
+            ),
+            custom_metric_filter(
+                error_information="Slurm Health Check Failure",
+                metric_name="Slurm Health Check Failure",
+                filter_pattern="Performing actions for health check type scheduled_events_check",
+            ),
+        ]
+
+        error_metric_dict = {
+            "Jobs Not Starting Errors": jobs_not_starting_errors,
+            "Unexpected EC2 Termination": compute_node_termination_errors,
+            "EC2 Events": job_issues_errors,
+        }
+
+        if self.config.head_node.custom_actions or self.config.scheduling.queues.custom_actions:
+            error_metric_dict["Custom Script Errors"].append(custom_script_errors)
```

Comment:
Have you tested your code with custom scripts?
I believe the code here should be `error_metric_dict["Custom Script Errors"]=(custom_script_errors)`

Reply:
yeah i changed it to   error_metric_dict.update({"Custom Script Errors": custom_script_errors})